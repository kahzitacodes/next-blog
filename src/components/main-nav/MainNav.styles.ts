import tw from 'tailwind-styled-components'
import * as ListStyles from '../list/List.styles'

export const Wrapper = tw.header``

export const Nav = tw.nav`
  items-center
  justify-between
  hidden
  lg:flex
`

export const Content = tw.div`
  flex
  z-50
  lg:hidden
`

export const MobileContainer = tw.div`
  fixed
  left-0
  top-0
  z-30
  w-full
  min-h-screen
  overflow-hidden
  bg-primary/90
  lg:hidden
  backdrop-blur-lg
  pt-[4rem]
  px-6
`

export const MobileNav = tw.nav`
  mx-auto
  flex

  items-center
  justify-center
  rounded-lg
  border
  border-gray-700
  bg-primary
  shadow-2xl
`

export const MobileList = tw(ListStyles.Wrapper)`
  w-full
  flex-col
  items-center
  justify-center
  px-5
`
export const MobileItem = tw(ListStyles.ListItem)`
  border-b
  border-gray-700
  py-4
  w-full
  last:border-b-0
  text-md
`
