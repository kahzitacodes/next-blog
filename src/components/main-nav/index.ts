export * from './MainNav'
export type { PropTypes as MainNavProps } from './MainNav.types'
