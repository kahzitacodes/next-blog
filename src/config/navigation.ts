import { NavItem } from '@/types'

export const navigationConfig: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'Posts',
    href: '/posts'
  },
  {
    title: 'About me',
    href: '/about'
  }
]
