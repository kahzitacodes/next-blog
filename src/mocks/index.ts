export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
