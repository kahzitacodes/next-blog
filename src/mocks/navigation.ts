import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
