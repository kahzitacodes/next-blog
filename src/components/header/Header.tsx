import { Logo } from '@/components'

import { PropTypes } from './Header.types'
import * as S from './Header.styles'

export const Header = ({ children }: PropTypes) => {
  return (
    <S.Container>
      <S.Content>
        <Logo />

        {children}
      </S.Content>
    </S.Container>
  )
}
