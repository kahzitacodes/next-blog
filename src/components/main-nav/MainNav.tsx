'use client'
import { ToogleButton } from './components/toogle-button'
import { PropTypes } from './MainNav.types'
import * as S from './MainNav.styles'
import { useMainNav } from '../hooks/useMainNav'
import { List } from '../list'

export const MainNav = ({ items }: PropTypes) => {
  const { handleToggleMenu, isMenuOpen } = useMainNav()

  return (
    <>
      <S.Nav>
        <List items={items} />
      </S.Nav>
      <S.Content>
        <ToogleButton isOpen={isMenuOpen} handleToggle={handleToggleMenu} />
      </S.Content>

      {isMenuOpen && (
        <S.MobileContainer>
          <S.MobileNav>
            <S.MobileList>
              {items?.map((item) => (
                <S.MobileItem key={item.title}>
                  <a href={item.href}>{item.title}</a>
                </S.MobileItem>
              ))}
            </S.MobileList>
          </S.MobileNav>
        </S.MobileContainer>
      )}
    </>
  )
}
