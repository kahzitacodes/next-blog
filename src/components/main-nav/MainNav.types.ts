import { NavItem } from '@/types/navigation'

export type PropTypes = {
  items: NavItem[]
}
