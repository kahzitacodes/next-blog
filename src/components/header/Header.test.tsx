import { render, screen } from '@/utils/tests/custom-render'
import { navigationMock } from '@/mocks/navigation'
import { Header } from './Header'
import { MainNav } from '../main-nav'

describe('<Profile />', () => {
  it('should render successfully', () => {
    const { container } = render(
      <Header>
        <MainNav items={navigationMock} />
      </Header>,
      {}
    )

    expect(screen.getByAltText(/Logo/i)).toBeInTheDocument()
    expect(screen.getByText(/home/i)).toBeInTheDocument()
    expect(screen.getByText(/about/i)).toBeInTheDocument()
    expect(screen.getByText(/contact/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
