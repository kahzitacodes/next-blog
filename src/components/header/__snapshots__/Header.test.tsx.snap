// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render successfully 1`] = `
<header
  class="fixed z-10 flex h-16 w-full items-center bg-primary/50 border-b border-gray-950/50 backdrop-blur-md transition-all duration-300 ease-out"
>
  <div
    class="flex h-full w-full items-center justify-between bg-primary/50 px-6"
  >
    <a
      class=""
      href="/"
    >
      <img
        alt="Logo"
        data-nimg="1"
        decoding="async"
        height="21"
        loading="lazy"
        src="/assets/images/logo.svg"
        style="color: transparent;"
        width="28"
      />
    </a>
    <nav
      class="items-center justify-between hidden lg:flex"
    >
      <ul
        class="flex gap-3"
      >
        <li
          class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
        >
          <a
            class="flex justify-center gap-3"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
        >
          <a
            class="flex justify-center gap-3"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
        >
          <a
            class="flex justify-center gap-3"
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="flex z-50 lg:hidden"
    >
      <button
        class="uflex transition-colors duration-300 hover:text-link"
        title="Open menu"
      >
        <svg
          class="lucide lucide-align-justify"
          data-testid="icon-open"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 12h18"
          />
          <path
            d="M3 18h18"
          />
          <path
            d="M3 6h18"
          />
        </svg>
      </button>
    </div>
  </div>
</header>
`;
